test/integration/sqlite_rpmdb_test.go

```diff
@@ -3,6 +3,8 @@ package integration
 import (
 	"testing"
 
+	_ "modernc.org/sqlite"
```

Comment:
this registers the sqlite driver only in test code, which wont get compiled into the final syft binary (which means the test will pass but syft will be unable to open sqlite RPMDBs since there will not be a registered driver)

Reply:
I pushed code that keeps this import, but I've moved it to the main package, this way library users can choose their own sqlite driver (we shouldn't be registering drivers that have side effects on the stdlib within our lib)

Reply:
I pushed another commit that keeps the import in main and this integration test (since this package does not use main it wouldn't have a sqlite driver).